src/foundry/foundry.js/clientDocuments/combat.d.ts

```diff
@@ -16,7 +18,202 @@ declare global {
+     *                  default `{}`
+     */
+    rollAll(options?: InitiativeOptions): this['rollInitiative'];
+
+    /**
+     * Roll initiative for all non-player actors who have not already rolled
+     * @param options - Additional options forwarded to the Combat.rollInitiative method
+     *                  default `{}`
+     */
+    rollNPC(options?: InitiativeOptions): this['rollInitiative'];
+
+    /**
+     * Assign initiative for a single Combatant within the Combat encounter.
+     * Update the Combat turn order to maintain the same combatant as the current turn.
+     * @param id - The combatant ID for which to set initiative
+     * @param value - A specific initiative value to set
+     */
+    setInitiative(id: string, value: number): Promise<void>;
+
+    /** Return the Array of combatants sorted into initiative order, breaking ties alphabetically by name. */
+    setupTurns(): InstanceType<ConfiguredDocumentClass<typeof Combatant>>[];
+
+    /** Begin the combat encounter, advancing to round 1 and turn 1 */
+    startCombat(): Promise<this>;
+
+    /**
+     * Define how the array of Combatants is sorted in the displayed list of the tracker.
+     * This method can be overridden by a system or module which needs to display combatants in an alternative order.
+     * By default sort by initiative, next falling back to name, lastly tie-breaking by combatant id.
+     */
+    protected _sortCombatants(
+      a: InstanceType<ConfiguredDocumentClass<typeof Combatant>>,
+      b: InstanceType<ConfiguredDocumentClass<typeof Combatant>>
+    ): number;
+
+    protected _onCreate(data: this['data']['_source'], options: DocumentModificationOptions, userId: string): void;
+
+    protected _onUpdate(
+      changed: DeepPartial<this['data']['_source']> & Record<string, unknown>,
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    protected _onDelete(options: DocumentModificationOptions, userId: string): void;
+
+    protected _onCreateEmbeddedDocuments(
+      embeddedName: string,
+      documents: foundry.abstract.Document<any, any>[],
+      result: Record<string, unknown>[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    protected _onUpdateEmbeddedDocuments(
+      embeddedName: string,
+      documents: foundry.abstract.Document<any, any>[],
+      result: Record<string, unknown>[],
+      options: DocumentModificationOptions,
+      userId: string
+    ): void;
+
+    protected _onDeleteEmbeddedDocuments(
+      embeddedName: string,
+      documents: foundry.abstract.Document<any, any>[],
+      result: string[],
+      options: DocumentModificationContext,
+      userId: string
+    ): void;
+
+    /** @deprecated since 0.8.0 */
+    getCombatant(id: string): this['combatants']['get'];
+
+    /** @deprecated since 0.8.0 */
+    createCombatant(
+      data: Array<Record<string, unknown>>,
```

Comment:
this is Combatant constructor data (which can be accessed via `Parameters<Combatant['data']['_initializeSource']>[0]`:
```suggestion
      data: Parameters<Combatant['data']['_initializeSource']>[0],
```